### PR TITLE
chore(ci): add azure pipeline ci checks

### PR DESCRIPTION
**Pull request by alewin**
Some CI improvements

- build check
- lint & formatting check
- test check

In the future we can also manage changelogs and releases via CI, I've prepared something on Analytics, but I'd like to write a proposal document and discuss it with the teams, before integrating it here 

after merge this PR I link the pipeline file to azure

### CODE REVIEW

**Comment by alewin** [comment]
waiting https://github.com/ogcio/building-blocks-sdk/pull/61 before merging it

File: .azure/pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  batch: true
+  branches:
+    include:
+      - main
+
+pr:
+  branches:
+    include:
+      - "*"
+
```
Comment (andreacappadona17) [inline .azure/pipeline.yaml:11]: Add the shared resources needed to use the shared templates

```suggestion
resources:
  repositories:
    - repository: pipeline-templates
      type: github
      name: ogcio/building-blocks-pipelines
      ref: main
      endpoint: ogcio

```

**Review by andreacappadona17 — APPROVED** [review]
LGTM!

@SamSalvatico @matteolc how does this look to you?

**Comment by SamSalvatico** [comment]
> LGTM!
> 
> @SamSalvatico @matteolc how does this look to you?

fine for me, thanks

File: vitest.config.mts [hunk 2]
```diff
@@ -4,7 +4,7 @@ export default defineConfig({
   test: {
     reporters: "default",
     coverage: {
-      reporter: ["text"],
+      reporter: ["text", "cobertura"],
```
Comment (matteolc) [inline vitest.config.mts:7]: Is `cobertura` intended?

**Pull request merged**